csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -291,7 +291,7 @@ private static bool IsDefaultValue(FieldDescriptor descriptor, object value) {
       return descriptor.FieldType switch {
         FieldType.Bool => (bool)value == false,
         FieldType.Bytes => (ByteString)value == ByteString.Empty,
-        FieldType.String => (string)value == "",
+        FieldType.String => ((string)value).Length == 0,
```

Comment:
This *should* be fine, as a field with a FieldType of string should never have a null value.